### PR TITLE
Feature Enhancement: Snip request through telegram

### PR DESCRIPTION
**Pull request by MerlionRock**
Re-Created due to conflicts

Added new feature:
Snipping through telegram, enable it in config, Sniper Task "mode": "telegram"

Also fixed the problem of not able to find Pokemons without EncounterID and SpawnPointID.

Fixes#5863

### CODE REVIEW

File: pokemongo_bot/cell_workers/sniper.py [hunk 1]
```diff
@@ -406,12 +449,15 @@ def work(self):
         else:
             self.bot.sniper_disabled_global_warning = False
             targets = []
+            target = {}
```
Comment (solderzzc) [inline pokemongo_bot/cell_workers/sniper.py]: Seems nowhere to use 'target' in the following code.

Comment (MerlionRock) [inline pokemongo_bot/cell_workers/sniper.py]: Noted. Will delete it off. I changed this files too many times and lost track what I did.

Comment (solderzzc) [inline pokemongo_bot/cell_workers/sniper.py]: Got it.

Comment (MerlionRock) [inline pokemongo_bot/cell_workers/sniper.py]: Done :)

**Comment by solderzzc** [comment]
👍

**Pull request merged**